app/models/user.rb

```diff
@@ -30,6 +32,8 @@ def restriction_list
   def self.from_omniauth(response)
     find_or_initialize_by(email: response['info']['email']) do |user|
       user.name = response['info']['name']
+      user.username = response['info']['email'].split('@').first if user.username.nil?
```

Comment:
Still feels odd exposing the users email.. Like on forums there's usually an option whether or not you want your email public.